src/reader/error_impl.rs

```diff
@@ -0,0 +1,101 @@
+
+use std::io;
+use std::borrow::Cow;
+use std::fmt;
+use std::error;
+
+use common::{Position, TextPosition};
+
+use super::{Error, ErrorKind};
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{} {}", self.pos, self.msg())
+    }
+}
+
+impl Position for Error {
+    #[inline]
+    fn position(&self) -> TextPosition { self.pos }
+}
+
+impl Error {
+    /// Returns a reference to a message which is contained inside this error.
+    #[inline]
+    pub fn msg(&self) -> &str {
+    	use super::ErrorKind::*;
+    	match self.kind {
+    		UnexpectedEof => &"Unexpected EOF",
+    		Utf8( ref reason ) => error_description( reason ),
+    		Io( ref io_error ) => error_description( io_error ),
+    		Syntax( ref msg ) => msg.as_ref(),
+    	}
+    }
+
+    pub fn kind( &self ) -> &ErrorKind { &self.kind }
+}
+
+impl error::Error for Error {
+    #[inline]
+    fn description(&self) -> &str { self.msg() }
+}
+
+impl<'a, P, M> From<( &'a P, M )> for Error where P: Position, M: Into<Cow<'static, str>> {
+	fn from( orig: (&'a P, M) ) -> Self {
+		Error{
+			pos: orig.0.position(),
+			kind: ErrorKind::Syntax( orig.1.into() )
+		}
+	}
+}
+
+impl From<::util::CharReadError> for Error {
```

Comment:
It think it is better to add `use util;` than to add a global namespace qualifier everywhere. Or even `use util::CharReadError`.